### PR TITLE
use `SignerDID` in offering.Verify() and rfq.Verify() methods

### LINKED ISSUE #23: check to make sure RFQ was signed by same DID as `Metadata.From`
[relevant code](https://github.com/TBD54566975/tbdex-go/blob/91c7ed04d7e29d95151f43cdc7117967c6e7de65/tbdex/rfq/rfq.go#L63-L66 )

`web5-go` supports this now as of `v0.16.0` thanks to @jiyoontbd ! so we can bump and skerp

### PR DESCRIPTION
**Pull request by jiyoonie9**
## What changed?
- Using `decoded.SignerDID` to compare with `message.metadata.from` field to ensure who signed the DID matches what's in the message / resource signature
- Wrote tests for this check
- Removing `from` field during `rfq/offering.create()` - message.metadata.from - if it's set during create(), it gets overwritten during sign() anyway. https://github.com/TBD54566975/tbdex-go/issues/29 will track combining create and sign into one method.

## Which issue(s) does this close? 
see linked issue to this PR

## How did you test this change?
- wrote unit tests

---
Not included in this PR, but an interesting way to hotswap methods being called (might want to do this in future test cases)

say you have a function like this
```go
func (o *Offering) Verify() error {
	decoded, err := tbdex.VerifySignature(o, o.Signature)
	if err != nil {
		return fmt.Errorf("failed to verify Offering signature: %w", err)
	}
        if decoded.SignerDID.URI != o.ResourceMetadata.From {
		return fmt.Errorf("signer: %s does not match message metadata from: %s", decoded.SignerDID.URI, o.ResourceMetadata.From)
	}
```

and you want to test the second if returns the error you expect. 

You can swap out `decoded, err := tbdex.VerifySignature(o, o.Signature)` for `decoded, err := VerifySignatureFunc(o, o.Signature)` and introduce this new file

```go
package offering

import (
	"github.com/TBD54566975/tbdex-go/tbdex"
	"github.com/tbd54566975/web5-go/jws"
)

var VerifySignatureFunc = func(o *Offering, signature string) (*jws.Decoded, error) {
	return tbdex.VerifySignature(o, signature)
}
```

and now you can hotswap the function like this in your test
```go

func Test_VerifyFailed() {
        originalFunc := offering.VerifySignatureFunc
	defer func() { offering.VerifySignatureFunc = originalFunc }() // resets the function

	offering.VerifySignatureFunc = func(o *offering.Offering, signature string) (*jws.Decoded, error) {
	did, _ := didjwk.Create()
	decoded := &jws.Decoded{
		SignerDID: did.DID,
	}
	return decoded, nil
}
// make your assertions here for the test failing the way you expect it to fail

}
```

### CODE REVIEW

File: tbdex/offering/offering_test.go [hunk 1]
```diff
@@ -101,3 +101,55 @@ func TestUnmarshal_Invalid(t *testing.T) {
...
+		),
+		"60000.00",
+	)
+
+	assert.NoError(t, err)
+
+	err = o.Sign(bearerDID)
+	assert.NoError(t, err)
+
+	o.Signature = "invalid"
```
Comment (jiyoonie9) [inline tbdex/offering/offering_test.go:152]: i'd like to use some kind of mock lib so i can have a `jws.Decoded` struct of my own returned for the `decoded, err := jws.Decode(o.Signature)` and then i can test the conditions checking the SignerDid.uri value against the metadata.from value

have not found a good solution for this yet. asked chad but they are providing solutions that aren't quite what i'm looking for

File: tbdex/rfq/rfq.go [hunk 2]
```diff
@@ -60,10 +61,14 @@ func (r *RFQ) Verify(privateDataStrict bool) error {
 		return fmt.Errorf("failed to verify RFQ signature: %w", err)
 	}
 
-	// TODO add check when decoded.SignerDID is implemented
-	// if decoded.SignerDID != r.MessageMetadata.From {
-	// 	return errors.New("signer: %w does not match message metadata from: %w", decoded.Header.SignerDID, r.MessageMetadata.From)
-	// }
+	decoded, err := jws.Decode(r.Signature)
```
Comment (phoebe-lew) [inline tbdex/rfq/rfq.go]: Don't need to call this again, `tbdex.VerifySignature` returns the decoded jws so you can just check against that directly by updating ln 59 to:

```
decoded, err := tbdex.VerifySignature(r, r.Signature)
```

File: tbdex/rfq/rfq_test.go [hunk 3]
```diff
@@ -439,3 +439,32 @@ func TestRFQ_Verify_FailsBadSignature(t *testing.T) {
 	err = rfq.Verify(false)
 	assert.Error(t, err)
 }
+
+
+func TestRFQ_Verify_InvalidSignature(t *testing.T) {
```
Comment (phoebe-lew) [inline tbdex/rfq/rfq_test.go]: Fixing the naming conventions for this in another PR:

```suggestion
func TestVerify_InvalidSignature(t *testing.T) {
```

File: tbdex/offering/offering.go [hunk 4]
```diff
@@ -149,10 +150,13 @@ func (o *Offering) Verify() error {
 		return fmt.Errorf("failed to verify Offering signature: %w", err)
 	}
 
-	// TODO once implemented in web5-go
-	// if jws.SignerDID != o.From {
-	// 	return errors.New("SignerDID: %w does not equal Offering.From: %w", jws.SignerDID, o.From)
-	// }
+	decoded, err := jws.Decode(o.Signature)
```
Comment (phoebe-lew) [inline tbdex/offering/offering.go]: Same as for RFQ, 
`decoded, err := tbdex.VerifySignature(o, o.Signature)`
above and you don't need to call `jws.Decode(...)` again

**Review by KendallWeihe — APPROVED** [review]

File: tbdex/rfq/signature_verifier.go [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package rfq
+
+import (
+	"github.com/TBD54566975/tbdex-go/tbdex"
+	"github.com/tbd54566975/web5-go/jws"
+)
+
+var VerifySignatureFunc = func(r *RFQ, signature string) (*jws.Decoded, error) {
```
Comment (phoebe-lew) [inline tbdex/rfq/signature_verifier.go]: what's the purpose of extracting this?

Comment (jiyoonie9) [reply tbdex/rfq/signature_verifier.go]: i was using it so i can hotswap the behavior i want this func to return during test. but it is not needed anymore so i removed it

**Pull request merged**